cmd/allocator/main.go

```diff
@@ -67,11 +122,30 @@ func main() {
 		ClientCAs:  caCertPool,
 	}
 	srv := &http.Server{
-		Addr:      ":" + port,
+		Addr:      ":" + sslPort,
 		TLSConfig: cfg,
+		// add http OC metrics (opencensus.io/http/server/*)
+		Handler: &ochttp.Handler{
+			Handler: httpsMux,
+		},
 	}
 
-	err = srv.ListenAndServeTLS(tlsDir+"tls.crt", tlsDir+"tls.key")
+	go func() {
+		var err error
+		lock := sync.Mutex{}
+		// force a pod restart if the https server exits.
+		health.AddLivenessCheck("allocator-https", func() error {
+			lock.Lock()
+			defer lock.Unlock()
+			return err
+		})
+		exitErr := srv.ListenAndServeTLS(tlsDir+"tls.crt", tlsDir+"tls.key")
+		lock.Lock()
+		err = exitErr
+		lock.Unlock()
+		logger.WithError(err).Fatal("allocation service crashed")
+	}()
+	err = http.ListenAndServe(":8080", http.DefaultServeMux)
```

Comment:
Can you add documentation to the code explaining why this listener on port 8080 is needed?